### ISSUE TITLE
detect conjugacies useful in BNP

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#547]
1) detect Dirichlet + categorical and beta + categorical 
(this should be easy since a categorical is just a multinomial with a single sample and I thought we actually already handled this...)

2) detect beta stick-breaking prior + categorical outcome (Chris has experience with stick-breaking so will look into this)
```
ksi_i ~ dcat(w)
w_k = z_k \prod_{l < k} (1-z_l)
z_k ~ dbeta
```

**Comment by danielturek** [comment]
@paciorek Let me know if this gets to a stage where my help or code review would be useful.  Otherwise, I'll consider it off my plate, at least for now.

**Comment by paciorek** [comment]
Case 1 is resolved by PR #578. 
Will be looking into case 2.

**Comment by paciorek** [comment]
This has been handled and we will address additional conjugacies as they arise in Claudia's work.
